bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Caret.java

```diff
@@ -396,15 +404,7 @@ void restoreIMEFont () {
 public void setBounds (int x, int y, int width, int height) {
 	checkWidget();
 	int zoom = getZoom();
-	setBoundsInPixels(DPIUtil.autoScaleUp(x, zoom), DPIUtil.autoScaleUp(y, zoom), DPIUtil.autoScaleUp(width, zoom), DPIUtil.autoScaleUp(height, zoom));
-}
-
-void setBoundsInPixels (int x, int y, int width, int height) {
-	boolean samePosition = this.x == x && this.y == y;
 	boolean sameExtent = this.width == width && this.height == height;
-	if (samePosition && sameExtent && isCurrentCaret()) return;
-	this.x = x;
-	this.y = y;
```

Comment:
Why must this be deferred to the `setBoundsInPixels` method call at the method's end instead of setting those values here?

Reply:
It is set in setBoundsInPixels where it was set earlier indeed.